istioctl/cmd/analyze.go

```diff
@@ -120,11 +129,7 @@ istioctl experimental analyze -k -d false
 				return err
 			}
 
-			for _, m := range messages {
-				fmt.Fprintf(cmd.OutOrStdout(), "%v\n", m.String())
-			}
-
-			return nil
+			return outputAndSetError(cmd.OutOrStdout(), messages)
```

Comment:
I sort of want to leave the "output" part out of this function, since we're likely to add more verbose output in several places and it'd be nice to handle that in a consistent way and it'd get awkward to pass it all to this function. IMO that's worth the extra loop through messages. (Not something I'd block on, and this isn't a one-way door, so take it for what it's worth to you).